apps/admin_api/test/admin_api/v1/controllers/blockchain/mint_controller_test.exs

```diff
@@ -0,0 +1,287 @@
+
+      response =
+        request("/token.mint", %{
+          id: token.id,
+          amount: 1_000_000 * token.subunit_to_unit
+        })
+
+      assert_mint(:existing_token, token, response)
+    end
+
+    test_with_auths "mints an existing ERC20 token with string amount", context do
+      enable_blockchain(context)
+      token = insert(:internal_blockchain_token)
+
+      response =
+        request("/token.mint", %{
+          id: token.id,
+          amount: "100000000"
+        })
+
+      assert_mint(:existing_token, token, response)
+    end
+
+    test_with_auths "mints an existing ERC20 token with a big number", context do
+      enable_blockchain(context)
+      token = insert(:internal_blockchain_token)
+
+      response =
+        request("/token.mint", %{
+          id: token.id,
+          amount: @big_number
+        })
+
+      mint = Mint |> Repo.all() |> Enum.at(0)
+
+      assert response["success"]
+      assert response["data"]["object"] == "mint"
+      assert Mint.get(response["data"]["id"]) != nil
+      assert mint != nil
+      assert mint.amount == @big_number
+      assert mint.token_uuid == token.uuid
+    end
+
+    test_with_auths "fails to mint an existing ERC20 token with amount = nil", context do
+      enable_blockchain(context)
+      token = insert(:internal_blockchain_token)
+
+      response =
+        request("/token.mint", %{
+          id: token.id,
+          amount: nil
+        })
+
+      refute response["success"]
+      assert response["data"]["code"] == "client:invalid_parameter"
+      assert response["data"]["description"] == "Invalid parameter provided."
+    end
+
+    test_with_auths "fails to mint a locked existing ERC20 token", context do
+      enable_blockchain(context)
+      token = insert(:internal_blockchain_token, locked: true)
+
+      response =
+        request("/token.mint", %{
+          id: token.id,
+          amount: "100000000"
+        })
+
+      mint = Mint |> Repo.all() |> Enum.at(0)
+      refute response["success"]
+      assert response["data"]["code"] == "token:is_locked"
+      assert response["data"]["description"] == "Minting is not allowed for this token."
+      assert mint == nil
+    end
+
+    test_with_auths "fails to mint an existing ERC20 token with mint amount sent as string",
+                    context do
+      enable_blockchain(context)
+      token = insert(:internal_blockchain_token)
+
+      response =
+        request("/token.mint", %{
+          id: token.id,
+          amount: "abc"
+        })
+
+      refute response["success"]
+      assert response["data"]["object"] == "error"
+      assert response["data"]["code"] == "client:invalid_parameter"
+
+      assert response["data"]["description"] ==
+               "Invalid parameter provided. String number is not a valid number: 'abc'."
+    end
+
+    test_with_auths "fails to mint an existing ERC20 token with mint amount == 0", context do
```

Comment:
Do we want the option to mint zero tokens - i.e. pass in zero as a possible value for `amount`? @T-Dnzt @mederic-p @DanielaIvanova

Reply:
If we mint with 0 - it is possible, the transaction will be mined, fees will be taken, call will be executed and it will mint 0 tokens, which from owner's perspective would be pointless.

Reply:
Sorry tagged this in the wrong place. Deployment not minting.